scripts/tcl_commands/routing.tcl

```diff
@@ -26,16 +26,43 @@ proc global_routing {args} {
 proc global_routing_or {args} {
     TIMER::timer_start
     set ::env(SAVE_DEF) $::env(fastroute_tmp_file_tag).def
-    try_catch openroad -exit $::env(SCRIPTS_DIR)/openroad/or_route.tcl |& tee $::env(TERMINAL_OUTPUT) $::env(fastroute_log_file_tag).log
+    try_catch openroad -exit $::env(SCRIPTS_DIR)/openroad/or_route.tcl |& tee $::env(TERMINAL_OUTPUT) $::env(fastroute_log_file_tag).log   
     if { $::env(DIODE_INSERTION_STRATEGY) == 3 } {
-      set ::env(DIODE_INSERTION_STRATEGY) 0
-      set_def $::env(SAVE_DEF)
-      try_catch openroad -exit $::env(SCRIPTS_DIR)/openroad/or_route.tcl |& tee $::env(TERMINAL_OUTPUT) $::env(fastroute_log_file_tag)_post_antenna.log
-      set ::env(DIODE_INSETION_STRATEGY) 3
+        set iter 2
+        set prevDEF1 $::env(SAVE_DEF)
+	set prevDEF2 $::env(SAVE_DEF)
+	set prevAntennaVal [exec grep "#Antenna violations:" $::env(fastroute_log_file_tag).log -s | tail -1 | sed -r "s/.*\[^0-9\]//"]
+        set_def $::env(SAVE_DEF)
+	while {$iter <= $::env(GLB_RT_MAX_DIODE_INS_ITERS) && $prevAntennaVal > 0} {
+            set ::env(SAVE_DEF) $::env(fastroute_tmp_file_tag)_$iter.def
+            set replaceWith "INSDIODE$iter"
+            try_catch sed -i -e "s/ANTENNA/$replaceWith/g" $::env(CURRENT_DEF)
```

Comment:
If the cell name is "ANTENNA" in the technology, which I believe is the case for some other technology I came across, this will fail.

Reply:
I'll modify it to (`ANTENNA_` to `INSDIODE$iter_`), the problem is that FastRoute uses the keyword `ANTENNA_idx` for insertion so from one iteration to the next 2 cells will be called `ANTENNA_0`, for example. That's why I'm renaming them.

Reply:
That doesn't solve the problem; you need make sure you're changing the instance name not the instance type from within the COMPONENTS section only. (e.g., `- instance_name instance_type....`). If a net or a pin for some reason has the string "ANTENNA_" within, it will inadvertently change.

Reply:
Roger that.

Reply:
bbc2de3b0cc6f79451d40909717abe3054f4be05